### PR TITLE
docs: Update instructions to install latest Docker

### PR DESCRIPTION
**Pull request by chavafg**
This commit adds instructions on how to install Docker 17
It also specifies to the user that Swarm only works with
Docker 1.12.1.

Signed-off-by: Salvador Fuentes <salvador.fuentes@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
qa-failed

**Comment by chavafg** [comment]
qa-passed

**Comment by jcvenegas** [comment]
@rcaballeromx  could you review this PR ?

File: documentation/Installing-Clear-Containers-on-Centos-7.md [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+distributions mainly due to the fact that it does not provide current versions
+of gcc and glibc.
+
+Hence, this guide explains how to install updated packages in `/usr/local`,
+to avoid conflicting with the base packages.
 
-Hence, this guide explains how to install updated packages in `/usr/local`, to avoid conflicting with the base packages.
+Clear containers supports latest version of Docker CE (currently 17.05), with
+the exception of Swarm. If you want to use Swarm, you should install Docker
+version 1.12.1.
```
Comment (amshinde) [inline documentation/Installing-Clear-Containers-on-Centos-7.md]: @chavafg What has broken with 17.05?

File: documentation/Installing-Clear-Containers-on-Centos-7.md [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
-Installing Clear Containers on CentOS is slightly more difficult than on other distributions mainly due to the fact that it does not provide current versions of gcc and glibc.
+Installing Clear Containers on CentOS is slightly more difficult than on other
+distributions mainly due to the fact that it does not provide current versions
+of gcc and glibc.
+
+Hence, this guide explains how to install updated packages in `/usr/local`,
+to avoid conflicting with the base packages.
 
-Hence, this guide explains how to install updated packages in `/usr/local`, to avoid conflicting with the base packages.
+Clear containers supports latest version of Docker CE (currently 17.05), with
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Centos-7.md]: s/supports latest/supports the latest/

File: documentation/Installing-Clear-Containers-on-Centos-7.md [hunk 3]
```diff
@@ -153,9 +166,21 @@ $ sudo yum install docker-engine-1.12.1-1.el7.centos.x86_64 docker-engine-selinu
...
+```
+$ sudo yum install -y yum-utils
+$ sudo yum-config-manager \
+    --add-repo \
+    https://download.docker.com/linux/centos/docker-ce.repo
+$ sudo yum makecache fast
+$ sudo yum install docker-ce
+```
 
+For more information on installing Docker please take a look at the
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Centos-7.md]: s/please take a look at the/please refer to/ sounds a little better.

File: documentation/Installing-Clear-Containers-on-Fedora.md [hunk 4]
```diff
@@ -1,4 +1,12 @@
-# Install the Clear Containers runtime on Fedora 25
+# Install the Clear Containers runtime on Fedora
+
+## Introduction
+
+Clear Containers 2.1 is available for Fedora versions **24** and **25**.
+Clear containers supports latest version of Docker CE (currently 17.05), with
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Fedora.md]: s/Clear containers supports latest version/Clear Containers supports the latest version/.

File: documentation/Installing-Clear-Containers-on-Fedora.md [hunk 5]
```diff
@@ -1,4 +1,12 @@
...
+the exception of Swarm. If you want to use Swarm, you should install Docker
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Fedora.md]: s/you should/you must/

File: documentation/Installing-Clear-Containers-on-Fedora.md [hunk 6]
```diff
@@ -27,6 +41,22 @@ Install docker engine:
...
+```
+$ sudo dnf -y install dnf-plugins-core
+$ sudo dnf config-manager \
+    --add-repo \
+    https://download.docker.com/linux/fedora/docker-ce.repo
+$ sudo dnf makecache fast
+$ sudo dnf install docker-ce
+```
+
+For more information on installing Docker please take a look at the
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Fedora.md]: s/please take a look at the/please refer to/ sounds a little better.

File: documentation/Installing-Clear-Containers-on-Ubuntu.md [hunk 7]
```diff
@@ -27,6 +32,28 @@ sudo apt-get install -y --allow-downgrades docker-engine=1.12.1-0~xenial
...
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+sudo add-apt-repository \
+	"deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+	$(lsb_release -cs) \
+	stable"
+sudo apt-get update
+sudo apt-get install docker-ce
+```
+
+For more information on installing Docker please take a look at the
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Ubuntu.md]: s/please take a look at the/please refer to/ sounds a little better.

File: documentation/Installing-Clear-Containers-on-Centos-7.md [hunk 8]
```diff
@@ -153,9 +166,21 @@ $ sudo yum install docker-engine-1.12.1-1.el7.centos.x86_64 docker-engine-selinu
 
 ```
 
-For more information on docker installation, see:
-- https://docs.docker.com/engine/installation/linux/centos/
+### Option 2) Docker 17
+
+**Warning:** Clear Conatiners 2.1 and Swarm will not work correctly on Docker 17.
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Centos-7.md]: s/Clear Conatiners/Clear Containers/.

File: documentation/Installing-Clear-Containers-on-Fedora.md [hunk 9]
```diff
@@ -27,6 +41,22 @@ Install docker engine:
 $ sudo dnf install docker-engine-1.12.1-1.fc24
 ```
 
+### Option 2) Docker 17
+
+**Warning:** Clear Conatiners 2.1 and Swarm will not work correctly on Docker 17.
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Fedora.md]: s/Clear Conatiners/Clear Containers/.

File: documentation/Installing-Clear-Containers-on-Ubuntu.md [hunk 10]
```diff
@@ -27,6 +32,28 @@ sudo apt-get install -y --allow-downgrades docker-engine=1.12.1-0~xenial
 sudo apt-mark hold docker-engine
 ```
 
+### Option 2) Docker 17
+
+**Warning:** Clear Conatiners 2.1 and Swarm will not work correctly on Docker 17.
```
Comment (jodh-intel) [inline documentation/Installing-Clear-Containers-on-Ubuntu.md]: s/Clear Conatiners/Clear Containers/.

File: documentation/Installing-Clear-Containers-on-Centos-7.md (hunk 1 above)
Comment (chavafg) [inline documentation/Installing-Clear-Containers-on-Centos-7.md]: This is the issue: #854, I think it has been broken since docker 1.12.6.

**Comment by chavafg** [comment]
@jodh-intel, fixes applied :) thanks.

**Comment by jodh-intel** [comment]
lgtm

**Comment by jodh-intel** [comment]
@rcaballeromx - could you take a peek at this please?

**Comment by rcaballeromx** [comment]
I am editing the changes now.

File: documentation/Installing-Clear-Containers-on-Centos-7.md (hunk 1 above)
Comment (amshinde) [inline documentation/Installing-Clear-Containers-on-Centos-7.md]: @chavafg I think the dns resolution was broken in 1.12.1 as well, we just did not have a test for it. Could you verify.

**Comment by chavafg** [comment]
lgtm for the change that @rcaballeromx made.

**Comment by chavafg** [comment]
qa-passed

**Pull request merged**